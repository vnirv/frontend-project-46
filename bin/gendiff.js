#!/usr/bin/env node
import '../src/gendiff.js';

