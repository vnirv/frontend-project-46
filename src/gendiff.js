import { program } from 'commander';
import { parceFiles } from './parcer.js';

import diff from './diff.js';

program
  .name('gendiff')
  .description('Compares two configuration files and shows a difference.')
  .version('1.0.0', '-v, --version', 'output the version number')
  .helpOption('-h, --help', 'output usage information')
  .option('-f, --format <format>', 'output format' )
  .argument('<tree>', 'inital tree path')
  .argument('<compareTree>', 'compare tree path');
  
program.parse();

const { format = 'json' } = program.opts()
const [path1, path2] = program.args;

const [tree, compareTree] = [path1, path2].map(parceFiles);

const result = diff(tree, compareTree, format);

console.log(result);