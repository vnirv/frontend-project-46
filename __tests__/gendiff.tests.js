
import gendiff from '../src/diff.js';


test('remplaced value', () => {
    const file1 = {name: 'meme', age: 0};
    const file2 = {name: 'meme', age: 2};
    const result = gendiff(file1, file2);
    
    expect(result).toMatchSnapshot();
});

test('added value', () => {
    const file1 = {name: 'meme'};
    const file2 = {name: 'meme', age: 1};
    const result = gendiff(file1, file2);
    expect(result).toMatchSnapshot();
});

test('deleted value', () => {
    const file1 = {name: 'meme', age: 0};
    const file2 = {name: 'meme'};
    const result = gendiff(file1, file2);
    expect(result).toMatchSnapshot();
});

test('deep structure', () => {
    const file1 = {name: 'meme', age: 0, state: {good: 'yes'}};
    const file2 = {name: 'meme', state: 'good'};
    const result = gendiff(file1, file2);
    expect(result).toMatchSnapshot();
});