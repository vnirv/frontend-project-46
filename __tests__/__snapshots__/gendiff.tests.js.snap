// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`added value 1`] = `
"{
      name: meme
    + age: 1
}"
`;

exports[`deep structure 1`] = `
"{
      name: meme
    - age: 0
    - state: {
        good: yes
    }
    + state: good
}"
`;

exports[`deleted value 1`] = `
"{
      name: meme
    - age: 0
}"
`;

exports[`remplaced value 1`] = `
"{
      name: meme
    - age: 0
    + age: 2
}"
`;
